Code/Tools/FBuild/FBuildCore/Graph/ObjectNode.cpp

```diff
@@ -821,6 +827,44 @@ const char * ObjectNode::GetObjExtension() const
 	return m_ObjExtensionOverride.Get();
 }
 
+// ContainsWarningsMSCL
+//------------------------------------------------------------------------------
+bool ObjectNode::ContainsWarningsMSCL(const char * data, uint32_t dataSize) const
+{
+	ASSERT( IsMSVC() );
+	bool containsWarnings = false;
+	static const char warningString[] = "warning";
+
+	if( ( data == nullptr ) || ( dataSize == 0 ) )
+	{
+		return containsWarnings;
+	}
+
+	const char * end = data + dataSize;
+	while( data < end )
+	{
+		// find the limits of the current line
+		const char * lineStart = data;
+		const char * lineEnd = lineStart;
+		while( lineEnd < end )
+		{
+			if( ( *lineEnd == '\r' ) || ( *lineEnd == '\n' ) )
+			{
+				break;
+			}
+			lineEnd++;
+		}
+
+		if( ( lineEnd - lineStart ) >= sizeof( warningString ) - 1 )
+		{
+			containsWarnings = containsWarnings || ( strstr( lineStart, warningString ) != nullptr );
```

Comment:
We could early out here which would also remove the need for the containsWarnings variable.